### PR TITLE
Cherry-pick improvements from kirkstone branch to master

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

**Comment by zyga** [comment]
Together with https://github.com/snapcore/meta-snapd/pull/38 this boots fine and reports strict confinement (`snap debug confinement`).

**Comment by bboozzoo** [comment]
I think this may need an update now, since there were changes that landed in kirkstone.

File: recipes-support/snapd/snapd-git.inc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+PR = "r0"
+
+SRC_URI = "git://github.com/snapcore/snapd.git;protocol=https;branch=release/2.58;destsuffix=git/"
```
Comment (bboozzoo) [inline recipes-support/snapd/snapd-git.inc:3]:
```suggestion
SRC_URI = "git://github.com/snapcore/snapd.git;protocol=https;branch=release/2.60;destsuffix=git/"
```
or maybe better add something like:
```
BASEPV = "2.60"
PV = "${BASEPV}+git${SRCPV}"
SRC_URI = "git://github.com/snapcore/snapd.git;protocol=https;branch=release/${BASEPV};destsuffix=git/"
```

Comment (bboozzoo) [reply recipes-support/snapd/snapd-git.inc:3]: which means that the _git recipe in kirkstone branch is broken too

Comment (zyga) [reply recipes-support/snapd/snapd-git.inc:3]: Updated https://github.com/snapcore/meta-snapd/pull/41 and here

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**